libvips/convolution/im_aconv.c

```diff
@@ -1005,9 +1005,7 @@ aconv_horizontal( Boxes *boxes, IMAGE *in, IMAGE *out )
 
 #define CLIP_UCHAR( V ) \
 G_STMT_START { \
-	if( (V) < 0 ) \
-		(V) = 0; \
-	else if( (V) > UCHAR_MAX ) \
+	if( (V) > UCHAR_MAX ) \
```

Comment:
The reason the warnings are generated, is because the clipping macros are used on unsigned types, were the < 0 check does nothing. If those macros were used on signed inputs the check would be correct.